pkg/proxy/endpoints.go

```diff
@@ -441,11 +451,17 @@ func (ect *EndpointChangeTracker) endpointsToEndpointsMap(endpoints *v1.Endpoint
 				isReady := true
 				isServing := true
 				isTerminating := false
-				isLocal := addr.NodeName != nil && *addr.NodeName == ect.hostname
+				isLocal := false
+				nodeName := ""
+				if addr.NodeName != nil {
+					isLocal = *addr.NodeName == ect.hostname
+					nodeName = *addr.NodeName
+				}
 				// Only supported with EndpointSlice API
 				zoneHints := sets.String{}
 
-				baseEndpointInfo := newBaseEndpointInfo(addr.IP, int(port.Port), isLocal, nil, isReady, isServing, isTerminating, zoneHints)
+				// Zone information is only supported with EndpointSlice API
+				baseEndpointInfo := newBaseEndpointInfo(addr.IP, nodeName, "", int(port.Port), isLocal, isReady, isServing, isTerminating, zoneHints)
```

Comment:
Will the nodeName change present a risk?  You'd have to NOT be using EPSlice to see this right?

Reply:
That is correct, so the risk is mitigated. I am also working on a PR to GA the gates so this code path should not be possible after that gets in.